src/main/java/com/arcbees/analytics/server/options/ServerOptionsCallback.java

```diff
@@ -68,17 +68,17 @@ public void onCallback(String options) {
     }
 
     @Override
-    public void putBoolean(String fieldName, boolean value) {
+    public synchronized void putBoolean(String fieldName, boolean value) {
         putText(fieldName, value ? "1" : "0");
     }
 
     @Override
-    public void putNumber(String fieldName, double value) {
+    public synchronized void putNumber(String fieldName, double value) {
         putText(fieldName, value + "");
     }
 
     @Override
-    public void putText(String fieldName, String value) {
+    public synchronized void putText(String fieldName, String value) {
         if (value == null) {
             options.remove(ProtocolTranslator.getFieldName(fieldName));
```

Comment:
Ok think ServerAnalytics should be made RequestScoped as well.  And the
maps should be concurrent.

On Fri, 16 Oct 2015 at 12:20 Thomas Broyer <notifications@github.com> wrote:

> In
> src/main/java/com/arcbees/analytics/server/options/ServerOptionsCallback.java
> <https://github.com/ArcBees/universal-analytics/pull/44#discussion_r42228259>
> :
>
> >          putText(fieldName, value + "");
> >      }
> >
> >      @Override
> > -    public void putText(String fieldName, String value) {
> > +    public synchronized void putText(String fieldName, String value) {
> >          if (value == null) {
> >              options.remove(ProtocolTranslator.getFieldName(fieldName));
>
> Yup: ServerAnalytics.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ArcBees/universal-analytics/pull/44/files#r42228259>.
>